nimble/controller/src/ble_ll_sched.c

```diff
@@ -826,6 +827,217 @@ ble_ll_sched_slave_new(struct ble_ll_conn_sm *connsm)
     return rc;
 }
 
+int
+ble_ll_sched_sync_reschedule(struct ble_ll_sched_item *sch,
+                             uint32_t anchor_point, uint32_t window_widening,
+                             int8_t phy_mode)
+{
+    int rc;
+    os_sr_t sr;
+    struct ble_ll_sched_item *start_overlap;
+    struct ble_ll_sched_item *end_overlap;
+    struct ble_ll_sched_item *entry;
+    struct ble_ll_conn_sm *tmp;
+
+    /* Set schedule start and end times */
+    sch->start_time = anchor_point - g_ble_ll_sched_offset_ticks;
+
+    sch->start_time -= (os_cputime_usecs_to_ticks(window_widening) + 1);
+    sch->remainder = 0;
+
+    /* TODO For now assume max sync packet, make expected data size
+     * configurable
+     */
+    sch->end_time = sch->start_time + os_cputime_usecs_to_ticks(
+                                        ble_ll_pdu_tx_time_get(256, phy_mode));
+
+    /* Better be past current time or we just leave */
+    if ((int32_t)(sch->start_time - os_cputime_get32()) < 0) {
+        return -2;
+    }
+
+    /* We have to find a place for this schedule */
+    OS_ENTER_CRITICAL(sr);
+
+    if (ble_ll_sched_overlaps_current(sch)) {
```

Comment:
This function used to check only if `sch` overlaps CONNECTION. Maybe this should be extended with SYNC, otherwise we break current sync interval?